pkg/services/auth/authimpl/external_session_store.go

```diff
@@ -0,0 +1,244 @@
+package authimpl
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+
+	"github.com/grafana/grafana/pkg/infra/db"
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/services/auth"
+	"github.com/grafana/grafana/pkg/services/secrets"
+)
+
+var _ auth.ExternalSessionStore = (*Store)(nil)
+
+type Store struct {
+	sqlStore       db.DB
+	secretsService secrets.Service
+	tracer         tracing.Tracer
+}
+
+func ProvideExternalSessionStore(sqlStore db.DB, secretService secrets.Service, tracer tracing.Tracer) auth.ExternalSessionStore {
```

Comment:
Nit: pretty sure we don't need to export these anymore

Reply:
Done